mitmproxy/flow/export.py

```diff
@@ -126,17 +137,21 @@ class WebsiteUser(HttpLocust):
             max_wait = 3000
 """).strip()
 
-    components = map(lambda x: quote(x, safe=""), flow.request.path_components)
-    file_name = "_".join(components)
-    name = re.sub('\W|^(?=\d)', '_', file_name)
-    url = flow.request.scheme + "://" + flow.request.host + "/" + "/".join(components)
+    components = [urllib.parse.quote(c, safe="") for c in flow.request.path_components]
+    name = re.sub('\W|^(?=\d)', '_', "_".join(components))
     if name == "" or name is None:
         new_name = "_".join([str(flow.request.host), str(flow.request.timestamp_start)])
         name = re.sub('\W|^(?=\d)', '_', new_name)
+
+    url = flow.request.scheme + "://" + flow.request.host + "/" + "/".join(components)
```

Comment:
I honestly have no idea how Locust works, but this may have been written if only the path is required in the url and not the query parameters. That doesn't make much sense though :/